ContributorAgreement.md

```diff
@@ -1,29 +1,28 @@
-freud Contributor Agreement
```

Comment:
The changes to this file are fine, but this combination of changes seems somewhat odd. If we assume that nobody is ever rendering this file as markdown, then there's no reason to mark the first line as a header. If we assume that the lines are rendered, then explicitly numbering below is unnecessary because markdown does it for you, and it's generally better not to have to manually number things to avoid human error. Is there a reason you chose to make these changes (especially since this file is identical to the ones used in our group's other repos)?

Reply:
I'm changing this so that it _matches_ other group software. We applied an identical set of changes to HOOMD-blue and fresnel recently.

We _should_ assume this file is rendered as Markdown - it ends in `.md`. The re-numbering, while not strictly necessary for Markdown renderers, makes it more readable for plain text editors.